src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -579,6 +576,29 @@ public void terminateTask(String taskId, TerminateInfo terminateInfo) {
         }
     }
 
+
+    /**
+     * Sets the termination process for a task during termination.
+     * this method fetches the termination process from Camunda and sets it on the task.
+     *
+     * @param taskId        The ID of the task being terminated.
+     * @param task          The task resource to update with the termination process.
+     */
+    private void setTerminationProcessOnTerminateTask(String taskId, TaskResource task) {
```

Comment:
Please use a helper class, it's easy to test and validate and most importantly service classes should contain only top level service logic. I can see bunch of other methods that don't belong here but this will be done as a general clean-up ticket.